client/my-sites/email/inbox/new-mailbox-upsell/index.tsx

```diff
@@ -22,13 +22,17 @@ const NewMailboxUpsell = ( { domains }: { domains: ResponseDomain[] } ) => {
 	// By default, upsell CTA links to the email management landing page.
 	let upsellURL = emailManagement( selectedSiteSlug, null, null, { source: INBOX_SOURCE } );
 
+	let isTrialNow = false;
+
 	// User has one single domain, determine / email addition page URL based on subscribed email provider.
 	if ( domains.length === 1 ) {
 		const domainItem = domains[ 0 ];
 
 		let slug = '';
 		if ( hasTitanMailWithUs( domainItem ) ) {
 			slug = 'titan/new';
+
+			isTrialNow = isUserOnTitanTrial( domainItem );
```

Comment:
@nightnei We probably don't need to introduce this new function here. We can rely on [getTitanMailboxPurchaseCost( domainItem )](https://github.com/Automattic/wp-calypso/blob/38c37bbfe35b8c741c4e346637987ddc0fb3ab59/client/lib/titan/get-titan-mailbox-purchase-cost.js) to retrieve the purchase cost object, we can set the `isTrialNow` flag by checking if `getTitanMailboxPurchaseCost( domainItem ).amount` is `0`

@AllTerrainDeveloper Please correct me if I'm wrong on this one.

Reply:
It will really work, and a similar approach [was already used in two other places](https://github.com/Automattic/wp-calypso/pull/66567/files#diff-faaeae7cc3937862cc4b14995597ad17f60733a2168b82f2d60ed59ebefe35c8L87).
But from my point of view - it's not obvious that `amount === 0` means that it's a free trial, and for me, it looks like some type of hardcode that is copy-pasted in a few places. That's why I decided to create the function, to isolate comparison and make the codebase more readable and clear (I spent some time finding and being sure that `amount === 0` is indeed the correct comparison to check the free trial).

What do you think about it?

Reply:
Even better, if you wait a bit @olaseni has added selectors in https://github.com/Automattic/wp-calypso/pull/66153. So you will be able to call `isDomainEligibleForTitanFreeTrial()` here 😀.

Reply:
@stephanethomas I have checked @olaseni changes and `isDomainEligibleForTitanFreeTrial` do another chacking.
`isDomainEligibleForTitanFreeTrial` - means that user don't have Professional email yet, but they are allowed to use free trial.
And my function `isUserOnTitanTrial ` - means that user already took Professional email and now they are on free trial.

Reply:
Ah right, good point. I was just passing by but I should definitively have paid more attention 😅.

May I suggest renaming `isUserOnTitanTrial()` to `isUserOnTitanFreeTrial()` though? That way it's consistent with `isDomainEligibleForTitanFreeTrial()` and the rest of our code.

Reply:
@wongasy I feel your concern, but I think it's a good addition to our code base. I imagine myself in the situation of having to check some user accounts being on trial or not, and I just found it difficult and tricky. I think adding a new function will ease further developments.

The only thing I would propose is to split the refactor in two pull request. As I discussed with Vova in our onboarding session, breaking down the pull requests as much as possible would benefit from:

- Faster reviews
- Faster testing
- Faster deploys
- More secure changes
- Better tracking

Reply:
> Ah right, good point. I was just passing by but I should definitively have paid more attention 😅.

np 😉

> May I suggest renaming isUserOnTitanTrial() to isUserOnTitanFreeTrial() though? That way it's consistent with isDomainEligibleForTitanFreeTrial() and the rest of our code.

Totally agree with you, and actually, I already did it locally 😄